src/main/java/com/powsybl/openloadflow/NetworkCache.java

```diff
@@ -0,0 +1,367 @@
+                    ? injection.getTerminal().getBusBreakerView().getBus()
+                    : injection.getTerminal().getBusView().getBus();
+        }
+
+        private boolean onLoadUpdate(Load load, String attribute, Object oldValue, Object newValue) {
+            boolean found = false;
+            for (AcLoadFlowContext context : contexts) {
+                Bus bus = getBus(load, context);
+                if (bus != null) {
+                    LfNetwork lfNetwork = context.getNetwork();
+                    LfBus lfBus = lfNetwork.getBusById(bus.getId());
+                    if (lfBus != null) {
+                        if (attribute.equals("p0")) {
+                            LoadDetail loadDetail = load.getExtension(LoadDetail.class);
+                            if (parameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_CONFORM_LOAD && loadDetail != null) {
+                                LOGGER.warn("Balance type PROPORTIONAL_TO_CONFORM_LOAD with Load {} that has a LoadDetail extension: not supported", load.getId());
+                            }
+                            double oldP0 = ((double) oldValue) / PerUnit.SB;
+                            double newP0 = ((double) newValue) / PerUnit.SB;
+                            double oldUpdatedP0 = LfContingency.getUpdatedLoadP0(lfBus, parameters.getBalanceType(), oldP0, oldP0);
+                            double newUpdatedLoadP0 = LfContingency.getUpdatedLoadP0(lfBus, parameters.getBalanceType(), newP0, newP0);
+                            lfBus.getAggregatedLoads().setInitialized(false);
+                            lfBus.reInitLoadTargetP(lfBus.getInitialLoadTargetP() + newP0 - oldP0);
+                            double newLoadP = lfBus.getLoadTargetP() - oldUpdatedP0 + newUpdatedLoadP0;
+                            lfBus.setLoadTargetP(newLoadP);
+                            context.setNetworkUpdated(true);
+                            found = true;
+                            break;
+                        } else {
+                            throw new IllegalStateException("Unsupported load attribute: " + attribute);
+                        }
+                    }
+                }
+            }
+            if (!found) {
+                LOGGER.warn("Cannot update {} of load '{}'", attribute, load.getId());
+            }
+            return found;
+        }
+
+        private boolean onGeneratorUpdate(Generator generator, String attribute, Object oldValue, Object newValue) {
+            boolean found = false;
+            for (AcLoadFlowContext context : contexts) {
+                Bus bus = getBus(generator, context);
+                if (bus != null) {
+                    LfNetwork lfNetwork = context.getNetwork();
+                    LfBus lfBus = lfNetwork.getBusById(bus.getId());
+                    if (lfBus != null) {
+                        if (attribute.equals("targetV")) {
```

Comment:
Just a remark here: in case of a shared control, the change of only one targetV will impact the voltage control targetV, even if we have other generators controlling that point with a different targetV.

Reply:
yes right